### PR TITLE
Reset rotation and position on double click

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Reset the image rotation and position on double click.

## Where does this idea come from?

This idea comes from Plotly:
https://plotly.com/python/creating-and-updating-figures/

## Quick demo:

https://user-images.githubusercontent.com/17039389/164958731-34783f2e-55f2-4051-9ef2-860af0aaf77d.mov

### CODE REVIEW

File: src/Preview.tsx [hunk 1]
```diff
@@ -229,6 +229,13 @@ const Preview: React.FC<PreviewProps> = props => {
     ],
   );
 
+  const onDoubleClick: React.MouseEventHandler<HTMLBodyElement> = () => {
+    if (visible) {
+      setScale(1);
+      setPosition(initialPosition);
+    }
+  };
```
Comment (afc163) [inline src/Preview.tsx:237]:
```suggestion
  const onDoubleClick: React.MouseEventHandler<HTMLBodyElement> = () => {
    if (!visible) {
      return;
    }
    setScale(1);
    setPosition(initialPosition);
  };
```

Comment (harupy) [reply src/Preview.tsx]: @afc163 Thanks for the quick review! Fixed!

**Comment by harupy** [comment]
Is the `CI / coverage` check failure related to this PR? I was able to reproduce the same error on the master branch by running `npm run test`.

**Comment by afc163** [comment]
Rebase master to fix snapshot.

**Comment by harupy** [comment]
> Rebase master to fix snapshot.

Done!

File: src/Preview.tsx [hunk 2]
```diff
@@ -248,6 +256,7 @@ const Preview: React.FC<PreviewProps> = props => {
       passive: false,
     });
     const onKeyDownListener = addEventListener(window, 'keydown', onKeyDown, false);
+    const onDoubleClickLister = addEventListener(window, 'dblclick', onDoubleClick, false);
```
Comment (afc163) [inline src/Preview.tsx:259]: useEffect return 内要卸载掉这些 addEventListener。

File: src/Preview.tsx [hunk 3]
```diff
@@ -229,6 +229,14 @@ const Preview: React.FC<PreviewProps> = props => {
     ],
   );
 
+  const onDoubleClick: React.MouseEventHandler<HTMLBodyElement> = () => {
+    if (!visible) {
+      return;
+    }
+    setScale(1);
+    setPosition(initialPosition);
+  };
```
Comment (afc163) [inline src/Preview.tsx:238]: 加个 useCallBack，不然每次刷新都进下面的 useEffect 了。

Comment (harupy) [reply src/Preview.tsx]: 完毕!

File: src/Preview.tsx [hunk 4]
```diff
@@ -317,6 +325,7 @@ const Preview: React.FC<PreviewProps> = props => {
       >
         <img
           onMouseDown={onMouseDown}
+          onDoubleClick={onDoubleClick}
```
Comment (harupy) [inline src/Preview.tsx:328]: @afc163 What do you think about the current implementation?

File: src/Preview.tsx [hunk 5]
```diff
@@ -229,6 +229,14 @@ const Preview: React.FC<PreviewProps> = props => {
     ],
   );
 
+  const onDoubleClick = () => {
+    if (!visible) {
+      return;
```
Comment (harupy) [inline src/Preview.tsx:234]:
```suggestion
    if (visible) {
      ...
    }
```

Can we use the initial approach to keep the test coverage 100%?

**Comment by harupy** [comment]
@afc163 Could you take another look?

File: src/Preview.tsx (hunk 5 above)
Comment (afc163) [reply src/Preview.tsx:234]: That is OK.

Comment (harupy) [reply src/Preview.tsx]: Got it, pushed a commit!

**Comment by harupy** [comment]
Interesting. The coverage check complained `ReferenceError: mount is not defined` but `mount` is defined.

**Comment by harupy** [comment]
rebased on master

**Comment by harupy** [comment]
I was wrong. `mount` is not defined.

**Comment by harupy** [comment]
@afc163 Fixed tests!

File: src/Preview.tsx [hunk 6]
```diff
@@ -229,6 +229,13 @@ const Preview: React.FC<PreviewProps> = props => {
     ],
   );
 
+  const onDoubleClick = () => {
+    if (visible) {
```
Comment (afc163) [inline src/Preview.tsx:233]:
```suggestion
    if (visible && scale !== 1) {
```

**Comment by harupy** [comment]
@afc163 Any clue on why tests failed?

**Comment by afc163** [comment]
Reinstall lock files and node_modules and try again.

**Comment by harupy** [comment]
I forgot to pull the latest commit.

**Pull request merged**